pcsx2/GS/Renderers/Common/GSDevice.cpp

```diff
@@ -723,7 +723,40 @@ GSTexture* GSDevice::CreateTexture(int w, int h, int mipmap_levels, GSTexture::F
 	return FetchSurface(GSTexture::Type::Texture, w, h, levels, format, false, m_features.prefer_new_textures && !prefer_reuse);
 }
 
-void GSDevice::StretchRect(GSTexture* sTex, GSTexture* dTex, const GSVector4& dRect, ShaderConvert shader, bool linear)
+void GSDevice::StretchRect(GSTexture* sTex, const GSVector4& sRect, GSTexture* dTex, const GSVector4& dRect,
+	GSHWDrawConfig::ColorMaskSelector cms, ShaderConvert shader, bool linear)
+{
+	pxAssert((dTex && dTex->IsDepthStencil()) == HasDepthOutput(shader));
+	pxAssert(linear ? SupportsBilinear(shader) : SupportsNearest(shader));
+	pxAssert(HasVariableWriteMask(shader) || cms.wrgba == ShaderConvertWriteMask(shader));
```

Comment:
The function that allows a color mask should always `pxAssert(HasVariableWriteMask(shader))`, even if the color mask happens to be 0xf.  Other StretchRect calls shouldn't need to check this at all, since they don't allow specifying color masks.